tslearn/nonmyopic.py

```diff
@@ -0,0 +1,509 @@
+
+        Returns
+        -------
+        Vector : the predicted classes
+        """
+
+        Y_pred = []
+
+        for t in range(0, X.shape[0]):
+            new_classif, new_time, new_proba, new_cost = self.classification(
+                X[t], self.cost_function
+            )
+            Y_pred.append(new_classif)
+        return Y_pred
+
+    def predict_completed(self, X, Y):
+        """
+        Predicts the classes of the series of the test dataset and has also access the the true classes to establish
+        comparisons
+
+        Parameters
+        ----------
+        X: Array-like
+            The data set that needs to be predict
+        Y: Array-vector
+            The true classes
+
+        Returns
+        -------
+        float : average time of prediction
+        float : variance of the time of prediction
+        float : AUC
+        float : average cost
+        """
+        Y_pred = []
+        time_predictions = []
+        pred_proba = []
+        cost_average = 0
+
+        for current_serie in range(0, X.shape[0]):
+            new_classif, new_time, new_proba, new_cost = self.classification(
+                X[current_serie], self.cost_function
+            )
+            Y_pred.append(new_classif)
+            time_predictions.append(new_time)
+            pred_proba.append(new_proba)
+            cost_average = cost_average + new_cost
+        average_pred, variance_pred = self.avg_prediction(time_predictions)
+        pred_proba = np.reshape(np.asarray(pred_proba), (X.shape[0], 2))
+        auc = roc_auc_score(Y, pred_proba[:, 1])
+        cost_average = cost_average / X.shape[0]
+        return average_pred, variance_pred, auc, cost_average
+
+    def graphic(self, path, name, prediction_time=None):
+        plt.clf()
+        for current_cluster in range(0, self.number_cluster):
+            for current_class in range(0, self.number_classes_):
+                for current_hat_class in range(0, self.number_classes_):
+                    if current_class != current_hat_class:
+                        sum_prob = [
+                            self.pyck_[current_class][current_cluster]
+                            * item[current_class][current_hat_class]
+                            for item in self.pyhatyck_[
+                                "pyhatycks{0}".format(current_cluster)
+                            ]
+                        ]
+                        lab = "y={} yhat={} ck={}".format(
+                            current_class, current_hat_class, current_cluster
+                        )
+                        plt.plot(sum_prob, label=lab)
+        plt.legend()
+        if prediction_time is not None:
+            plt.title("The prediction time is " + str(prediction_time))
+        plt.savefig(os.path.join(path, name + ".png"), dpi=500)
+
+    def cost_time(self, t):
+        time_stamp = np.linspace(0, self.len_X_, num=self.len_X_)
+        return time_stamp[t] * self.cost_time_parameter
```

Comment:
Isn't the formula simply `t * self.cost_time_parameter`? I do not see what you are doing here.